composer/trainer/mosaic_fsdp.py

```diff
@@ -84,8 +84,8 @@ def get_mixed_precision(precision, mixed_precision='DEFAULT', keep_low_precision
         if mixed_precision == 'FULL':
             pass
         elif mixed_precision == 'DEFAULT':
-            reduce_dtype = get_torch_dtype(precision)
-            buffer_dtype = torch.float32
+            param_dtype = get_torch_dtype(precision)
+            buffer_dtype = get_torch_dtype(precision)
```

Comment:
Should we just make default PURE and get rid of DEFAULT? Because it seems like you should always use PURE?

Reply:
AMP is an industry standard and should be the `DEFAULT`.
PURE might work well with autocast_precision=bf16, but it might not work well with autocast_precision=fp16. These differences is not something we've tested so I wouldn't remove `DEFAULT`.